### PR TITLE
Support paths with + in List API

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
NSQueryParams doesn't encode `+` to `%2B`, which breaks the List API since it treats `+` as space. 

Note that other Storage APIs don't rely on query param encoding. There is already an integration tests that verifies that users can up and download files with `+` signs. 

Fixes b/170259897

### CODE REVIEW

File: FirebaseStorage/Tests/Unit/FIRStorageListTests.m [hunk 1]
```diff
@@ -152,6 +152,42 @@ - (void)testListWithPageSizeAndPageToken {
   [FIRStorageTestHelpers waitForExpectation:self];
 }
 
+- (void)testPercentEncodesPlusToken {
+  XCTestExpectation *expectation = [self expectationWithDescription:@"testPercentEncodesPlusToken"];
+  NSURL *expectedURL = [NSURL URLWithString:@"https://firebasestorage.googleapis.com/v0/b/bucket/"
+                                            @"o?prefix=%2Bfoo/&delimiter=/"];
+
+  self.fetcherService.testBlock =
+      ^(GTMSessionFetcher *fetcher, GTMSessionFetcherTestResponse response) {
```
Comment (paulb777) [inline FirebaseStorage/Tests/Unit/FIRStorageListTests.m:161]: Can you avoid the pragma with an approach like #6693?

**Review by paulb777 — APPROVED** [review]
Please add a changelog

File: FirebaseStorage/Tests/Unit/FIRStorageListTests.m (hunk 1 above)
Comment (schmidt-sebastian) [reply FirebaseStorage/Tests/Unit/FIRStorageListTests.m:161]: I am getting an unused variable warning, and I don't see how the code you linked to doesn't generate the same warning. Is there any trick I need to employ?

**Review by schmidt-sebastian — COMMENTED** [review]
Sorry, forgot to push the Changelog. Added.

**Comment by paulb777** [comment]
There's something weird going on with the XCTest macros. In the Messaging test, they reference self but not in the Storage test - I can't see what about the projects would trigger the difference.

I also don't reproduce the warning when I remove the #pragma's in the Storage test.

This probably isn't worth this much time - so I'm fine to merge as is.

**Comment by schmidt-sebastian** [comment]
Thanks @paulb777!

**Pull request merged**